Upstream-Status: Backport

* Fixes "linux/dvb/dmx.h:137:2: error: unknown type name '__u64'" when compiled with -std=c99.

Signed-off-by: Andreas Oberritter <obi@opendreambox.org>

--- linux-2.6.18/include/asm-mips/types.h.orig	2012-05-16 13:50:07.238037869 +0200
+++ linux-2.6.18/include/asm-mips/types.h	2012-05-16 13:50:20.462038208 +0200
@@ -34,9 +34,9 @@
 
 #else
 
-#if defined(__GNUC__) && !defined(__STRICT_ANSI__)
-typedef __signed__ long long __s64;
-typedef unsigned long long __u64;
+#if defined(__GNUC__)
+__extension__ typedef __signed__ long long __s64;
+__extension__ typedef unsigned long long __u64;
 #endif
 
 #endif
